src/StargateEventListener/Handlers/StargateEventHandlers.ts

```diff
@@ -0,0 +1,269 @@
+    events: any[],
+    nodeManagementContract: string,
+    beat: Beat,
+    txId: string,
+): NodeDelegatedEventData[] => {
+    const nodeDelegatedEvents: NodeDelegatedEventData[] = []
+
+    for (const event of events) {
+        if (isNodeDelegatedEvent(event, nodeManagementContract)) {
+            const decoded = decodeNodeDelegatedEvent(event)
+            if (decoded) {
+                const eventData: NodeDelegatedEventData = {
+                    ...decoded,
+                    blockNumber: beat.number,
+                    txId,
+                    contractAddress: nodeManagementContract,
+                }
+
+                nodeDelegatedEvents.push(eventData)
+                logFoundEvent(decoded)
+            }
+        }
+    }
+
+    return nodeDelegatedEvents
+}
+
+/**
+ * Check if an event is a NodeDelegated event from the correct contract
+ */
+const isNodeDelegatedEvent = (event: any, nodeManagementContract: string): boolean => {
+    return (
+        event.address?.toLowerCase() === nodeManagementContract.toLowerCase() &&
+        event.topics?.[0]?.toLowerCase() === NODE_DELEGATED_EVENT_SIGNATURE.toLowerCase()
+    )
+}
+
+/**
+ * Log details of a found NodeDelegated event
+ */
+const logFoundEvent = (decoded: Omit<NodeDelegatedEventData, "blockNumber" | "txId" | "contractAddress">): void => {
+    debug(
+        ERROR_EVENTS.STARGATE,
+        `Found NodeDelegated event: nodeId=${decoded.nodeId}, ` +
+            `delegatee=${decoded.delegatee}, isDelegated=${decoded.isDelegated}`,
+    )
+}
+
+/**
+ * Decode a NodeDelegated event from VeChain event logs
+ */
+const decodeNodeDelegatedEvent = (event: {
+    topics: string[]
+    data: string
+}): Omit<NodeDelegatedEventData, "blockNumber" | "txId" | "contractAddress"> | null => {
+    try {
+        if (event.topics[0]?.toLowerCase() !== NODE_DELEGATED_EVENT_SIGNATURE.toLowerCase()) {
+            return null
+        }
+
+        const decoded = new abi.Event(NODE_DELEGATED_EVENT_ABI).decode(event.data, event.topics)
+
+        return {
+            nodeId: decoded.nodeId.toString(),
+            delegatee: decoded.delegatee,
+            isDelegated: decoded.isDelegated,
+        }
+    } catch (err) {
+        error(ERROR_EVENTS.STARGATE, "Failed to decode NodeDelegated event:", err)
+        return null
+    }
+}
+
+/**
+ * Process a single NodeDelegated event
+ */
+const processNodeDelegatedEvent = async (eventData: NodeDelegatedEventData, refetchStargateData: () => void) => {
+    try {
+        debug(ERROR_EVENTS.STARGATE, "Processing NodeDelegated event:", {
+            nodeId: eventData.nodeId,
+            delegatee: eventData.delegatee,
+            isDelegated: eventData.isDelegated,
+            blockNumber: eventData.blockNumber,
+            txId: eventData.txId,
+        })
+
+        // Trigger refetch of Stargate data
+        // This will automatically update UI components (StakedCard, AccountFiatBalance, etc.)
+        refetchStargateData()
```

Comment:
I believe you should refetch only if the node owner is a wallet in VeWorld, otherwise this will be triggered every time a random user does it